File: Easy/sede/meta_sededefaultview/meta_sededefaultview.cs
/*
    Easy
    Copyright (C) 2020 Università degli Studi di Catania (www.unict.it)
    This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.
    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.
    You should have received a copy of the GNU General Public License
    along with this program.  If not, see <http://www.gnu.org/licenses/>.
*/

﻿using System;
using System.Data;
using metadatalibrary;
using metaeasylibrary;
//$CustomUsing$


namespace meta_sededefaultview
{
    public class Meta_sededefaultview : Meta_easydata 
	{
        public Meta_sededefaultview(DataAccess Conn, MetaDataDispatcher Dispatcher) :
            base(Conn, Dispatcher, "sededefaultview") {
				Name = "Sedi";
			EditTypes.Add("default");
			ListingTypes.Add("default");
			//$EditTypes$
        }

		private string[] mykey = new string[] {"idsede"};

		public override string[] primaryKey() {
			return mykey;
		}

		//$SetDefault$

		//$Get_New_Row$

		//$IsValid$

		public override void DescribeColumns(DataTable T, string ListingType) {
			base.DescribeColumns(T, ListingType);

			foreach (DataColumn C in T.Columns) {
				DescribeAColumn(T, C.ColumnName, "", -1);
			}
			int nPos = 1;
					
			switch (ListingType) {
				case "default": {
						DescribeAColumn(T, "title", "Denominazione", nPos++);
						DescribeAColumn(T, "sede_idsedemiur", "Identificativo MIUR", nPos++);
						DescribeAColumn(T, "sede_address", "Indirizzo", nPos++);
						DescribeAColumn(T, "sede_cap", "CAP", nPos++);
						DescribeAColumn(T, "geo_city_title", "Comune", nPos++);
						break;
					}
					//$DescribeAColumn$
			}
		}

		public override string GetSorting(string ListingType) {
			switch (ListingType) {
				case "default": {
						return "title asc ";
					}
				//$GetSorting$
			}
			return base.GetSorting(ListingType);
		}

		public override string GetStaticFilter(string ListingType) {
			switch (ListingType) {
				case "default": {
						return "idreg = (SELECT top(1) idreg from istitutoprinc)";
						break;
					}
				//$GetStaticFilter$
			}
			return base.GetStaticFilter(ListingType);
		}
		
		//$CustomCode$
    }
}

File: Easy/Portale/client/app_segreterie/metadata/sede/meta_sededefaultview.js
(function() {

    var MetaData = window.appMeta.MetaSegreterieData;

    function meta_sededefaultview() {
        MetaData.apply(this, ["sededefaultview"]);
        this.name = 'meta_sededefaultview';
    }

    meta_sededefaultview.prototype = _.extend(
        new MetaData(),
        {
            constructor: meta_sededefaultview,
			superClass: MetaData.prototype,

			describeColumns: function (table, listType) {
				var nPos=1;
				var self = this;
				_.forEach(table.columns, function (c) {
					self.describeAColumn(table, c.name, '', null, -1, null);
				});
				switch (listType) {
					default:
						return this.superClass.describeColumns(table, listType);
					case 'default':
						this.describeAColumn(table, 'title', 'Denominazione', null, 20, 1024);
						this.describeAColumn(table, 'sede_idsedemiur', 'Identificativo MIUR', null, 30, null);
						this.describeAColumn(table, 'sede_address', 'Indirizzo', null, 40, 100);
						this.describeAColumn(table, 'sede_cap', 'CAP', null, 50, 20);
						this.describeAColumn(table, 'geo_city_title', 'Comune', null, 60, 65);
//$objCalcFieldConfig_default$
						break;
//$objCalcFieldConfig$
				}
				return appMeta.Deferred("describeColumns").resolve();
			},


			//$setCaptions$

			primaryKey: function () {
				return ["idsede"];
			},


			//$isValidFunction$

			//$getStaticFilter$

			getSorting: function (listType) {
				switch (listType) {
					case "default": {
						return "title asc ";
					}
					//$getSortingin$
				}
				return this.superClass.getSorting(listType);
			}

        });

    window.appMeta.addMeta('sededefaultview', new meta_sededefaultview('sededefaultview'));

	}());
